### DISCUSSION TITLE
pulse energy weapon won't fire

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#8299] [Q&A] [answered]
not sure if it is a bug or I'm doing something wrong, but I have a aerie variant, replaced the missiles with one heavy laser and one plasma cannon. both laser torrents with pulse torrent.
I have two power cores installed (Geode and Pebble)
when I go into battle and fire, only the one heavy laser gun fires

this is the branch I'm testing: https://github.com/daggs1/endless-sky/tree/full_features
based on 8e4eb04f481cd5a9e37f6de32c95dc7dffdff161

**Comment by vitalchip** [comment, marked as answer]
You don't have enough energy. Add a battery.

The problem is how much instantaneous energy you have, Geode + Pebble can provide 13.4 in a frame, the Heavy Laser consumes 1, so it can fire.

The Plasma Cannon needs 30 per shot.. you don't have it. Likewise the Pulse Turrets need 24 per shot each.

You'll have enough energy across a second to fire all weapons but the energy will be required in spikes which you can't provide. Even a Supercapacitor should be able to smooth it out enough.

**Reply by daggs1** [reply]
facepalm, I got a Geode instead of a Boulder...

shame on me...
if I add a battery, how long it will suffice in battles?

**Reply by vitalchip** [reply]
That really depends on what else is using energy at the time.

There is enough available energy across a second to run the weapons continuously but whether you can run those and your engines and shield regeneration and *whatever* needs more information.

**Reply by daggs1** [reply]
both shield and engines are hai

**Reply by daggs1** [reply]
replaced the two reactors with a boulder, added a hai fissure battery pack, all works. thanks.
